src/Npgsql/NpgsqlSchema.cs

```diff
@@ -16,33 +18,38 @@ namespace Npgsql
     static class NpgsqlSchema
     {
         public static DataTable GetSchema(NpgsqlConnection conn, string? collectionName, string?[]? restrictions)
+            => GetSchemaAsync(conn, collectionName, restrictions, false).GetAwaiter().GetResult();
+
+        public static ValueTask<DataTable> GetSchemaAsync(NpgsqlConnection conn, string? collectionName, string?[]? restrictions, bool async = true, CancellationToken cancellationToken = default)
```

Comment:
As this is a very perf-insensitive API that will have to contact the database for most values, this could just return Task instead of ValueTask - optimizing the cases where we don't need to contact PostgreSQL for the synchronous case seems like a bit much.